tools/test-backend

```diff
@@ -89,16 +89,15 @@ if __name__ == "__main__":
 
     def rewrite_arguments(search_key):
         # type: (str) -> None
-        for file_name in os.listdir(zerver_test_dir):
-            if file_name.endswith(".py"):
-                filepath = zerver_test_dir + file_name
-                if os.path.isdir(filepath):
-                    continue
-                for line in open(filepath):
-                    if search_key in line:
-                        new_suite = filepath.replace(".py", ".") + suite
-                        args[args.index(suite)] = new_suite
-                        return
+        for root, dirs, files_names in os.walk(zerver_test_dir, topdown=False):
+            for file_name in files_names:
+                if file_name[0].isalnum() and file_name.endswith(".py"):
```

Comment:
Also, I'd consider inverting the check and using a `continue` statement to decrease the level of nesting, since this is getting really deeply nested.